homeassistant/components/filesize/sensor.py

```diff
@@ -76,32 +75,22 @@ class Filesize(SensorEntity):
     def __init__(self, path: str) -> None:
         """Initialize the data object."""
         self._path = path  # Need to check its a valid path
-        self._size: int | None = None
-        self._last_updated: str | None = None
         self._attr_name = slugify(path.split("/")[-1]).strip().capitalize()
-        self._attr_unique_id = path
+        self._attr_extra_state_attributes = {}
```

Comment:
There is not need to initialise it.
```suggestion
```

Reply:
It's not initialized in the base so it is needed to have here unless you use property

Reply:
I think it can stay uninitialised/`None` until the update has run.

Reply:
No, you need to initiate it if you want to use it later such as in update.
Failed this too many times already as when you want to use the _attr attributes you need to carefully see which ones init by themselves or when you need to do it.

Reply:
It needs to be initialised if `self._attr_extra_state_attributes` gets updated later.
But in this integration, you are not updating the attributes, you are setting them to a new dict inside the `update` routine.

Looking at `trafikverket_train.sensor`, it is the same logic and `self._attr_extra_state_attributes` is not initialised.
Looking at `tile.device_tracker`, it is initialised because the update routine updates it.

Reply:
Sorry, you're right. Wasn't considering it could return None --> Will change.
Can I blame I got covid at the moment 😅